src/bentoml/bentos.py

```diff
@@ -419,3 +425,103 @@ def containerize(bento_tag: Tag | str, **kwargs: t.Any) -> bool:
     except Exception as e:  # pylint: disable=broad-except
         logger.error("Failed to containerize %s: %s", bento_tag, e)
         return False
+
+
+def serve(
+    bento: str,
+    production: bool = False,
+    port: int = BentoMLContainer.http.port.get(),
+    host: str = BentoMLContainer.http.host.get(),
+    type: str = "HTTP",
+    api_workers: int | None = BentoMLContainer.api_server_workers.get(),
+    backlog: int = BentoMLContainer.api_server_config.backlog.get(),
+    reload: bool = False,
+    working_dir: str | None = None,
+    ssl_certfile: str | None = None,
+    ssl_keyfile: str | None = None,
+    ssl_ca_certs: str | None = None,
+    # HTTP-specific args
+    ssl_keyfile_password: str | None = None,
+    ssl_version: int | None = None,
+    ssl_cert_reqs: int | None = None,
+    ssl_ciphers: str | None = None,
+    # GRPC-specific args
+    enable_reflection: bool = BentoMLContainer.grpc.reflection.enabled.get(),
+    enable_channelz: bool = BentoMLContainer.grpc.channelz.enabled.get(),
+    max_concurrent_streams: int
+    | None = BentoMLContainer.grpc.max_concurrent_streams.get(),
+) -> Server:
+    """Launch a BentoServer and returns a client that exposes all APIs defined in target service"""
+
+    if type not in ["HTTP", "GRPC"]:
+        raise ValueError('Server type must either be "HTTP" or "GRPC"')
+
+    args = [
+        str(shutil.which("bentoml")),
```

Comment:
lets not do this. lets do
```python
"-m", "bentoml"
```
and pass the executable `sys.executable` to subprocess.Popen